CorsixTH/Lua/world.lua

```diff
@@ -474,6 +474,36 @@ local function isDiseaseUsableForNewPatient(self, disease, hospital)
   error("disease has neither visuals_id or non_visuals_id")
 end
 
+--! Private function to assess the shortest path between all spawn points and desks
+--!param desks (table) All staffed reception desks
+--!param spawns (table) All available spawn points
+--!return Optimal spawn point to the nearest reception desk or nil on failure
+function World:_findClosestSpawnToDesk(desks, spawns)
+  local best_spawn_point, distance
+  local function testNewSpawnPath(desk, spawn)
+    local new_dist = self:getPathDistance(desk.tile_x, desk.tile_y, spawn.x, spawn.y)
+    if not new_dist then return end
+    if not distance or new_dist < distance then
+      distance = new_dist
+      best_spawn_point = spawn
+    end
+  end
+
+  for _, spawn in ipairs(spawns) do
+    for _, desk in pairs(desks) do
```

Comment:
It's a list, right? 

Maybe also name it `staffed_desk` to express that property clearly.

Reply:
Tbh it should use `ipairs`, I forgot to change that in my late-night nitpick commit